src/plugins/expressions/common/expression_functions/specs/map_column.ts

```diff
@@ -0,0 +1,124 @@
+  name: string;
+  expression?: (datatable: Datatable) => Promise<boolean | number | string | null>;
+  copyMetaFrom?: string | null;
+}
+
+export const mapColumn: ExpressionFunctionDefinition<
+  'mapColumn',
+  Datatable,
+  MapColumnArguments,
+  Promise<Datatable>
+> = {
+  name: 'mapColumn',
+  aliases: ['mc'], // midnight commander. So many times I've launched midnight commander instead of moving a file.
+  type: 'datatable',
+  inputTypes: ['datatable'],
+  help: i18n.translate('expressions.functions.mapColumnHelpText', {
+    defaultMessage:
+      'Adds a column calculated as the result of other columns. ' +
+      'Changes are made only when you provide arguments.' +
+      'See also {alterColumnFn} and {staticColumnFn}.',
+    values: {
+      alterColumnFn: '`alterColumn`',
+      staticColumnFn: '`staticColumn`',
+    },
+  }),
+  args: {
+    id: {
+      types: ['string', 'null'],
+      help: i18n.translate('expressions.functions.mapColumn.args.idHelpText', {
+        defaultMessage:
+          'An optional id of the resulting column. When `null` the name/column argument is used as id.',
+      }),
+      required: false,
+      default: null,
+    },
+    name: {
+      types: ['string'],
+      aliases: ['_', 'column'],
+      help: i18n.translate('expressions.functions.mapColumn.args.nameHelpText', {
+        defaultMessage: 'The name of the resulting column.',
+      }),
+      required: true,
+    },
+    expression: {
+      types: ['boolean', 'number', 'string', 'null'],
+      resolve: false,
+      aliases: ['exp', 'fn', 'function'],
+      help: i18n.translate('expressions.functions.mapColumn.args.expressionHelpText', {
+        defaultMessage:
+          'A {CANVAS} expression that is passed to each row as a single row {DATATABLE}.',
+        values: {
+          CANVAS: 'canvas',
+          DATATABLE: '`datatable`',
+        },
+      }),
+      required: true,
+    },
+    copyMetaFrom: {
+      types: ['string', 'null'],
+      help: i18n.translate('expressions.functions.mapColumn.args.copyMetaFromHelpText', {
+        defaultMessage:
+          "If set, the meta object from the specified column id is copied over to the specified target column. Throws an exception if the column doesn't exist",
+      }),
+      required: false,
+      default: null,
+    },
+  },
+  fn: (input, args) => {
+    const expression = args.expression || (() => Promise.resolve(null));
+    const columnId = args.id != null ? args.id : args.name;
+
+    const columns = [...input.columns];
+    const rowPromises = input.rows.map((row) => {
+      return expression({
+        type: 'datatable',
+        columns,
+        rows: [row],
+      }).then((val) => ({
+        ...row,
+        [columnId]: val,
+      }));
+    });
+
+    return Promise.all(rowPromises).then((rows) => {
+      const existingColumnIndex = columns.findIndex(({ name }) => name === args.name);
+      const type = rows.length ? getType(rows[0][args.name]) : 'null';
```

Comment:
If id and name are both used, this lookup won't work. It should use `columnId` as well (probably worth a functional test(

Reply:
Fixed the issue. I did misinterpret it initially. There's a test already to cover this specific case.